clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -0,0 +1,340 @@
+        String missing = findMissing(MAX_MISSING_TO_LOG, response.responseData().keySet(),
+            sessionParts.keySet());
+        if (!missing.isEmpty()) {
+            log.warn("Session handler for node {} received a full fetch response " +
+                "that contained unexpected partition(s) {}", node, missing);
+            return false;
+        }
+        missing = findMissing(MAX_MISSING_TO_LOG, sessionParts.keySet(),
+            response.responseData().keySet());
+        if (!missing.isEmpty()) {
+            log.warn("Session handler for node {} received a full fetch response " +
+                "that did not contain expected partition(s) {}", node, missing);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Verify that the partitions in an incremental fetch response are contained in the session.
+     *
+     * @param response  The response.
+     * @return          True if the incremental fetch response partitions are valid.
+     */
+    private boolean verifyIncrementalFetchResponseParts(FetchResponse response) {
+        String missing = findMissing(MAX_MISSING_TO_LOG, response.responseData().keySet(),
+            sessionParts.keySet());
+        if (!missing.isEmpty()) {
+            log.warn("Session handler for node {} received an incremental fetch response " +
+                "that contained unexpected partition(s) {}", node, missing);
+            return false;
+        }
+        return true;
+    }
+
+    public boolean handleResponse(FetchResponse response) {
+        if (response.error() == Errors.NONE) {
+            log.debug("Session handler for node {} received FetchResponse for {} with session " +
+                    "id {} and {} partition(s).", node, nextMetadata,
+                response.sessionId(), response.responseData().size());
+            if (nextMetadata.isFull()) {
+                if (!verifyFullFetchResponseParts(response)) {
+                    nextMetadata = FetchMetadata.INITIAL;
+                    return false;
+                } else {
+                    if (response.sessionId() == INVALID_SESSION_ID) {
+                        nextMetadata = FetchMetadata.INITIAL;
+                    } else {
+                        nextMetadata = FetchMetadata.newIncremental(response.sessionId());
+                    }
+                    return true;
+                }
+            } else {
+                if (!verifyIncrementalFetchResponseParts(response)) {
+                    nextMetadata = nextMetadata.nextCloseExisting();
+                    return false;
+                } else {
+                    nextMetadata = nextMetadata.nextIncremental();
+                    return true;
+                }
+            }
+        } else if (response.error() == Errors.FETCH_SESSION_ID_NOT_FOUND) {
+            log.info("Session handler for node {} received FETCH_SESSION_ID_NOT_FOUND error on {}.",
```

Comment:
I am wondering if this can be lowered to `DEBUG` since it is handled internally.

Reply:
I think it makes sense to log since it's a pretty rare occurrence.  And if it does start happening a lot, that could indicate a problem.